lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -627,13 +575,22 @@ class ModuleEmitter : public EmitterBase {
   /// resolveKeywordConflict, to get a new name in case of conflict with
   /// keywords.
   StringRef getModuleName(StringAttr moduleName) {
-    auto entryIter = moduleNameTable.find(moduleName);
-    if (entryIter != moduleNameTable.end())
-      return entryIter->getSecond();
-    auto updatedName = resolveKeywordConflict(
-        moduleName.getValue(), usedModuleNames, nextGeneratedNameID);
-    moduleNameTable[moduleName] = updatedName;
-    return updatedName;
+    // TODO: The renaming is now performed in the RTLLegalizeNamesPass. This bit
+    // of code merely checks if the original implementation would have suggested
+    // a different renaming. Remove this once we're happy about the pre-pass.
+    auto actualName = moduleName.getValue();
+    if (!moduleNameTable.count(moduleName)) {
+      auto expectedName = resolveKeywordConflict(
+          moduleName.getValue(), usedModuleNames, nextGeneratedNameID);
+      if (actualName != expectedName) {
+        llvm::errs() << "ExportVerilog suggested \"" << expectedName
+                     << "\" instead of \"" << actualName << "\"\n";
+        assert(false && "ExportVerilog suggested a module renaming that should "
```

Comment:
Please change this into an emitOpError.  Library code like this shouldn't print to llvm::errs(), and ideally shouldn't abort on invalid input.